internal/component/configmap/configmap.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package configmap
+
+import (
+	"encoding/json"
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	"github.com/gardener/etcd-druid/internal/component"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetConfigMap indicates an error in getting the configmap resource.
+	ErrGetConfigMap druidv1alpha1.ErrorCode = "ERR_GET_CONFIGMAP"
+	// ErrSyncConfigMap indicates an error in syncing the configmap resource.
+	ErrSyncConfigMap druidv1alpha1.ErrorCode = "ERR_SYNC_CONFIGMAP"
+	// ErrDeleteConfigMap indicates an error in deleting the configmap resource.
+	ErrDeleteConfigMap druidv1alpha1.ErrorCode = "ERR_DELETE_CONFIGMAP"
+)
+
+const etcdConfigKey = "etcd.conf.yaml"
```

Comment:
Nit:
Is it possible to move this to `internal/common/constants.go`? is it defined as a constant in the sts package as well

Reply:
Thanks Aaron. Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/9ca05f252426a8986a1d40b5e367ea4b81f23054